src/tests/testHttpRequest.cc

```diff
@@ -121,36 +123,48 @@ TestHttpRequest::testIPv6HostColonBug()
-    CPPUNIT_ASSERT_EQUAL(SBuf("/foo"), aRequest->url.path());
-    CPPUNIT_ASSERT_EQUAL(AnyP::PROTO_HTTP, static_cast<AnyP::ProtocolType>(aRequest->url.getScheme()));
-
-    /* valid IPv6 address with port */
+    if (Ip::EnableIpv6) {
+        CPPUNIT_ASSERT(aRequest != nullptr);
+        CPPUNIT_ASSERT_EQUAL(AnyP::KnownPort(80), *aRequest->url.port());
+        CPPUNIT_ASSERT(aRequest->method == Http::METHOD_GET);
+        CPPUNIT_ASSERT_EQUAL(String("[2000:800::45]"), String(aRequest->url.host()));
+        CPPUNIT_ASSERT_EQUAL(SBuf("/foo"), aRequest->url.path());
+        CPPUNIT_ASSERT_EQUAL(AnyP::PROTO_HTTP, static_cast<AnyP::ProtocolType>(aRequest->url.getScheme()));
+    } else {
+        CPPUNIT_ASSERT(!aRequest);
+    }
+
+    /* bracketed IPv6 address with port */
     url = "http://[2000:800::45]:90/foo";
     aRequest = HttpRequest::FromUrl(url, mx, Http::METHOD_GET);
-    expected_port = 90;
-    CPPUNIT_ASSERT_EQUAL(expected_port, *aRequest->url.port());
-    CPPUNIT_ASSERT(aRequest->method == Http::METHOD_GET);
-    CPPUNIT_ASSERT_EQUAL(String("[2000:800::45]"), String(aRequest->url.host()));
-    CPPUNIT_ASSERT_EQUAL(SBuf("/foo"), aRequest->url.path());
-    CPPUNIT_ASSERT_EQUAL(AnyP::PROTO_HTTP, static_cast<AnyP::ProtocolType>(aRequest->url.getScheme()));
-
-    /* IPv6 address as invalid (bug trigger) */
+    if (Ip::EnableIpv6) {
+        CPPUNIT_ASSERT(aRequest != nullptr);
+        CPPUNIT_ASSERT_EQUAL(AnyP::KnownPort(90), *aRequest->url.port());
+        CPPUNIT_ASSERT(aRequest->method == Http::METHOD_GET);
+        CPPUNIT_ASSERT_EQUAL(String("[2000:800::45]"), String(aRequest->url.host()));
+        CPPUNIT_ASSERT_EQUAL(SBuf("/foo"), aRequest->url.path());
+        CPPUNIT_ASSERT_EQUAL(AnyP::PROTO_HTTP, static_cast<AnyP::ProtocolType>(aRequest->url.getScheme()));
+    } else {
+        CPPUNIT_ASSERT(!aRequest);
+    }
+
+    /* bracket-less IPv6 address without port (bug trigger) */
     url = "http://2000:800::45/foo";
     aRequest = HttpRequest::FromUrl(url, mx, Http::METHOD_GET);
-    expected_port = 80;
-    CPPUNIT_ASSERT_EQUAL(expected_port, *aRequest->url.port());
-    CPPUNIT_ASSERT(aRequest->method == Http::METHOD_GET);
-    CPPUNIT_ASSERT_EQUAL(String("[2000:800::45]"), String(aRequest->url.host()));
-    CPPUNIT_ASSERT_EQUAL(SBuf("/foo"), aRequest->url.path());
-    CPPUNIT_ASSERT_EQUAL(AnyP::PROTO_HTTP, static_cast<AnyP::ProtocolType>(aRequest->url.getScheme()));
+    if (Ip::EnableIpv6) {
+        CPPUNIT_ASSERT(aRequest != nullptr);
+        CPPUNIT_ASSERT_EQUAL(AnyP::KnownPort(80), *aRequest->url.port());
+        CPPUNIT_ASSERT(aRequest->method == Http::METHOD_GET);
+        CPPUNIT_ASSERT_EQUAL(String("[2000:800::45]"), String(aRequest->url.host()));
+        CPPUNIT_ASSERT_EQUAL(SBuf("/foo"), aRequest->url.path());
+        CPPUNIT_ASSERT_EQUAL(AnyP::PROTO_HTTP, static_cast<AnyP::ProtocolType>(aRequest->url.getScheme()));
+    } else {
+        CPPUNIT_ASSERT(!aRequest);
+    }
 }
```

Comment:
This method/test case should be renamed/split and expanded to handle more use cases, including:

* bracket-less IPv6 in GET requests with a port: `GET http://::1:80/`.
* bracketed IPv4 address, with and without a port
* bracketed domain name, with and without a port
* bracket-less IPv6 in CONNECT requests with IPv6 support disabled (at least until we merge CONNECT and GET host parsing);

IMO, all those tests are outside this PR scope (and some will require more primary code changes), but they can be added in this PR if others insist.